inference-engine/src/mkldnn_plugin/mkldnn_graph.h

```diff
@@ -200,6 +200,9 @@ class MKLDNNGraph {
     std::vector<MKLDNNNodePtr> graphNodes;
     std::vector<MKLDNNEdgePtr> graphEdges;
 
+    std::vector<MKLDNNNodePtr> constantGraphNodes;
+    std::vector<MKLDNNNodePtr> mutableGraphNodes;
```

Comment:
The problem I see with this approach: we create duplicate holders for graph nodes (in addition two graphNodes) which are not synchronized in any way. Do we have any good approach two encapsulate them somewhere to prevent others from using them (before SplitNodes() call)?
From the top of my mind I can suggest remove them from fields and pass to required functions as parameters. 
Otherwise we at least have to move them into private section and add detailed comment with usage insights. 

@maxnick any thoughts?

Reply:
We want to remove useless code in inference time and add it to load time to avoid overhead during inference.
If we would pass nodes to functions as parameters without class fields, we should extract constant nodes before `Infer` method but inside `InferImpl`. But it's inference anyway. So we won't have any improvements
So I just added comments for class fields

The nodes are shared_ptr, so we have some sync between graphNodes and constantGraphNodes, mutableGraphNodes (unfortunately, this is not about cases of adding or removing nodes)

Reply:
As far as I understand we call `ExtractConstantNodes()` after all possible transformations at the end of the `InitGraph()` procedure, so the graphNodes content should not be changed afterwards. Therefore such splitting is safe by the internal contract and graph initialization algorithm itself. But, to insure better reliability we can extract the  Replication and Transformation stages into another class. So that the new class become a kind of factory for constant graph object that may not be changed and can perform inference, input/output processing, data damping etc.